0-bootstrap/modules/jenkins/main.tf

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cicd_project_name       = format("%s-%s", var.project_prefix, "cicd")
+  jenkins_apis             = ["cloudkms.googleapis.com"]
+  impersonation_enabled_count = var.sa_enable_impersonation == true ? 1 : 0
+  activate_apis               = distinct(var.activate_apis)
+  jenkins_gce_fw_tags         = ["ssh-jenkins-agent"]
+}
+
+/******************************************
+  CICD project
+*******************************************/
+module "cicd_project" {
+  source                      = "terraform-google-modules/project-factory/google"
+  version                     = "~> 7.0"
+  name                        = local.cicd_project_name
+  random_project_id           = true
+  disable_services_on_destroy = false
+  folder_id                   = var.folder_id
+  org_id                      = var.org_id
+  billing_account             = var.billing_account
+  activate_apis               = local.activate_apis
+  labels                      = var.project_labels
+}
+
+resource "google_project_service" "jenkins_apis" {
+  for_each           = toset(local.jenkins_apis)
+  project            = module.cicd_project.project_id
+  service            = each.value
+  disable_on_destroy = false
+}
+
+/******************************************
+  Jenkins Agent GCE instance
+*******************************************/
+resource "google_service_account" "jenkins_agent_gce_sa" {
+  project      = module.cicd_project.project_id
+  account_id   = var.jenkins_sa_email
+  display_name = "CFT Jenkins Agent GCE custom Service Account"
+}
+
+resource "google_compute_instance" "jenkins_agent_gce_instance" {
+  project      = module.cicd_project.project_id
+  name         = var.jenkins_agent_gce_name
+  machine_type = "n1-standard-1"
+  zone         = "europe-west2-a"
```

Comment:
Might be best to parameterize with a default?

Reply:
Done